document-parsers/langchain4j-document-parser-apache-tika/src/main/java/dev/langchain4j/data/document/parser/apache/tika/ApacheTikaDocumentParser.java

```diff
@@ -48,22 +53,45 @@ public ApacheTikaDocumentParser() {
      * @param contentHandler Tika content handler. Default: {@link BodyContentHandler} without write limit
      * @param metadata       Tika metadata. Default: empty {@link Metadata}
      * @param parseContext   Tika parse context. Default: empty {@link ParseContext}
+     * @deprecated Use the constructor with suppliers for Tika components if you intend to use this parser for multiple files.
      */
+    @Deprecated
     public ApacheTikaDocumentParser(Parser parser,
                                     ContentHandler contentHandler,
                                     Metadata metadata,
                                     ParseContext parseContext) {
-        this.parser = getOrDefault(parser, AutoDetectParser::new);
-        this.contentHandler = getOrDefault(contentHandler, () -> new BodyContentHandler(NO_WRITE_LIMIT));
-        this.metadata = getOrDefault(metadata, Metadata::new);
-        this.parseContext = getOrDefault(parseContext, ParseContext::new);
+        this(() -> parser, () -> contentHandler, () -> metadata, () -> parseContext);
+    }
+
+    /**
+     * Creates an instance of an {@code ApacheTikaDocumentParser} with the provided suppliers for Tika components.
+     * If some of the suppliers are not provided ({@code null}), the defaults will be used.
+     *
+     * @param parserSupplier         Supplier for Tika parser to use. Default: {@link AutoDetectParser}
+     * @param contentHandlerSupplier Supplier for Tika content handler. Default: {@link BodyContentHandler} without write limit
+     * @param metadataSupplier       Supplier for Tika metadata. Default: empty {@link Metadata}
+     * @param parseContextSupplier   Supplier for Tika parse context. Default: empty {@link ParseContext}
+     */
+    public ApacheTikaDocumentParser(Supplier<Parser> parserSupplier,
+                                    Supplier<ContentHandler> contentHandlerSupplier,
+                                    Supplier<Metadata> metadataSupplier,
+                                    Supplier<ParseContext> parseContextSupplier) {
+        this.parserSupplier = parserSupplier;
+        this.contentHandlerSupplier = contentHandlerSupplier;
+        this.metadataSupplier = metadataSupplier;
+        this.parseContextSupplier = parseContextSupplier;
     }
 
     // TODO allow automatically extract metadata (e.g. creator, last-author, created/modified timestamp, etc)
 
     @Override
     public Document parse(InputStream inputStream) {
         try {
+            Parser parser = getOrDefault(parserSupplier, () -> DEFAULT_PARSER_SUPPLIER).get();
```

Comment:
minor: I would call `getOrDefault` in the constructor instead and do `parserSupplier.get()` here (same for the rest of fields)

Reply:
Done !